### ISSUE TITLE
Reactions show usernames instead of real names

### ISSUE THREAD

**Issue by lpilz** [RocketChat/Rocket.Chat.ReactNative#2152]
- Your Rocket.Chat Experimental app version: 4.7.0.17188
- Your Rocket.Chat server version: 3.2.2
- Device (or Simulator) you're running with: Samsung Galaxy Note 8

The setting `UI_use_real_name` is ignored in the app, usernames are always shown.

**Comment by djorkaeffalexandre** [comment]
Hey @lpilz, can you provide more info about when it happens at the thread replies?
With this we can create a checklist to fix all the points!
Thanks in advance.

**Comment by lpilz** [comment]
Hey, I'm currently trying to nail down the bug. What's for sure is that for reactions, it is probably not implemented. I just found out, that the username issue in threads is also occurring in the desktop version, so this might be a bug in the main repo somewhere. I'll focus the issue on reactions then :)

**Comment by lpilz** [comment]
FYI: The thread issue is known and will apparently be fixed with the coming thread rewrite:
https://github.com/RocketChat/Rocket.Chat/issues/15662
https://github.com/RocketChat/Rocket.Chat/issues/17519

**Comment by expert-group** [comment]
You are right @lpilz  It is displaying username. but if you want to display real name, you can change it on code easily.

**Comment by Kushagraw12** [comment]
Hey I am new here. Can I work on this issue?
Thanks!

**Comment by diegolmello** [comment]
@Kushagraw12 sure :)

**Comment by Kushagraw12** [comment]
@diegolmello  Thanks, can you please tell me from where should i start? In which file should i look for the issue?
Thanks!

**Comment by diegolmello** [comment]
@Kushagraw12 We should test if `UI_Use_Real_Name` is set [here](https://github.com/RocketChat/Rocket.Chat.ReactNative/blob/develop/app/containers/ReactionsModal.js#L65) and show the real name instead.

**Comment by Kushagraw12** [comment]
Hi! I have been trying for days now, and i keep running into an error while launching my the app locally. Can someone please help me with that?

![Screenshot (56)](https://user-images.githubusercontent.com/53997603/87819055-4318e080-c889-11ea-953a-9685a9a7848c.png)

This is the error that i am getting.
I have attached my android device to my system an even ran this command after that `adb -s <device name> reverse tcp:8081 tcp:8081`
I followed all the steps given here [Set-up React Native](https://reactnative.dev/docs/environment-setup)
Please help me with this. Thanks!

**Comment by diegolmello** [comment]
@Kushagraw12 
> Install the missing components using SDK manager in Android Studio

You probably are missing SDK 28.

**Comment by Kushagraw12** [comment]
> @Kushagraw12
> 
> > Install the missing components using SDK manager in Android Studio
> 
> You probably are missing SDK 28.

Ohh, so for that, I shoot up my android Studio and add SDK 28 manually from there right?
Or is there some other way to handle it?
Thanks!

**Comment by Kushagraw12** [comment]
@diegolmello  I added SDK 28 and tried to run the app again, but i am encountering a different error this time. I tried to google this error, but i could not understand the solution. Please help me out! Thanks! 

![Screenshot (57)](https://user-images.githubusercontent.com/53997603/87851225-37263080-c914-11ea-98fa-ac94ea739607.png)

**Comment by diegolmello** [comment]
It looks like you don't have a valid JDK.

**Comment by jdsatava127** [comment]
We are experiencing a related issue using Docker image 3.5.4. For us though, Real Name displays fine everywhere outside of threads (normal messages and reactions), but displays username in the threads. I was able to manually edit a message in Mongo and set the "u.name" value for a message in a thread and that allowed the Real Name to display. Is the thread UI possibly expecting this value to be set rather than looking up the Real Name on the fly?

**Comment by diegolmello** [comment]
@jdsatava127 can you open a separate issue with steps? thanks

**Comment by jdsatava127** [comment]
A new issue has been created here - https://github.com/RocketChat/Rocket.Chat/issues/18769

**Comment by lucasmidlhey** [comment]
this problem has been resolve?

**Comment by namaniiitm** [comment]
Hey @diegolmello can I take up this issue?

**Comment by Namanl2001** [comment]
Hey! @diegolmello can I take up this issue?

**Comment by Namanl2001** [comment]
Hey! @diegolmello can I take up this issue?

**Comment by diegolmello** [comment]
@Namanl2001 Some work was done on #2365 already.
We just need to refactor it to reuse the action sheet.

**Comment by rajoriaakash** [comment]
Hi, I am new to JavaScript, and this project.
I would really like to work on this issue. Could you please help me get a little overview of the code that would be related to this issue? Thanks!

**Comment by diegolmello** [comment]
Welcome, @rajoriaakash!
There's a PR open for this issue already on #2939.

**Comment by rajoriaakash** [comment]
Oh, ok. Thanks for replying.
Guess I'll just go through that to have a better understanding of code.

**Comment by rajoriaakash** [comment]
@diegolmello I have a doubt.
How can we access the `UI_Use_Real_Name` or `useRealName` in the `ReactionsModal` file. I am having a slight difficulty understanding the code. Could you please help? Thanks!
